include/boost/url/bnf/parse.hpp

```diff
@@ -16,58 +16,79 @@
 #include <boost/type_traits/make_void.hpp>
 #include <type_traits>
 
+/** @file
+
+   Parsing commands
+
+   This file contains the basic overloads for parsing BNF elements
+
+   @par Design of BNF elements
+
+   A BNF element implements the rules for parsing a BNF grammar rule and stores parsing results.
+
+   A BNF element:
+
+    - Is default constructible
+    - Holds the result, usually as a reference, of parsing
+    - Has a nested value_type
+    - Has `value_type& operator*() const noexcept`
+
+   Users should define a free function `parse` as a customization point defining how to parse their
+   BNF elements as part of the same architecture that might include arbitrary BNF elements in expressions.
+
+   These new function overloads may be defined in other namespaces. As with `std::swap`, the design relies
+   on Argument Dependent Lookup (ADL) to find these overloads.
```

Comment:
You have to put this in the exposition

Reply:
Where would you recommend it to go?

Reply:
> Where would you recommend it to go?

Well, somewhere is better than nowhere. That said, let me have a look...

Make a new section 4.0.BNF_parsing and dump it there. Don't forget to include the new qbk file.

Reply:
Still not moved

Reply:
OK then